src/vs/workbench/services/textfile/electron-browser/nativeTextFileService.ts

```diff
@@ -232,37 +231,8 @@ export class NativeTextFileService extends AbstractTextFileService {
 	}
 
 	private getEncodedReadable(value: string | ITextSnapshot, encoding: string, addBOM: boolean): VSBufferReadable {
-		const readable = this.snapshotToNodeReadable(typeof value === 'string' ? stringToSnapshot(value) : value);
-		const encoder = encodeStream(encoding, { addBOM });
-
-		const encodedReadable = readable.pipe(encoder);
-
-		return nodeStreamToVSBufferReadable(encodedReadable, addBOM && isUTFEncoding(encoding) ? { encoding } : undefined);
-	}
-
-	private snapshotToNodeReadable(snapshot: ITextSnapshot): Readable {
-		return new Readable({
-			read: function () {
-				try {
-					let chunk: string | null = null;
-					let canPush = true;
-
-					// Push all chunks as long as we can push and as long as
-					// the underlying snapshot returns strings to us
-					while (canPush && typeof (chunk = snapshot.read()) === 'string') {
-						canPush = this.push(chunk);
-					}
-
-					// Signal EOS by pushing NULL
-					if (typeof chunk !== 'string') {
-						this.push(null);
-					}
-				} catch (error) {
-					this.emit('error', error);
-				}
-			},
-			encoding: UTF8 // very important, so that strings are passed around and not buffers!
-		});
+		const snapshot = typeof value === 'string' ? stringToSnapshot(value) : value;
+		return toEncodeReadable(snapshot, encoding, { addBOM });
```

Comment:
The `ITextSnapshot` interface here do match `Readable` (but does not implement it) It is accidental or intentional?

Reply:
@gyzerok historically the `ITextSnapshot` interface was first and then I added `Readable`, I think it is fine to leave it since TS takes care of understanding the interface is implemented by structural type-check.

Reply:
I am wondering if it makes sense to separately change the code so that `ITextSnapshot implements Readable`. Not as part of this change, but in general. Otherwise thanks for the explanation!

Reply:
I think it is fine like this given TS takes care, but thanks for noting.